v23.1/fips-compliance.md

```diff
@@ -0,0 +1,355 @@
+
+- You can model your Dockerfile on the one that Cockroach Labs uses to produce the [FIPS-ready Docker image](https://github.com/cockroachdb/cockroach/blob/master/build/deploy/Dockerfile) for CockroachDB.
+- Your Dockerfile must install OpenSSL before it starts the `cockroach` binary.
+- FIPS mode must be enabled on the Docker host kernel before it can run containers with FIPS mode enabled. The FIPS-ready CockroachDB Docker image must run with FIPS mode enabled. To enable FIPS mode in the Docker host kernel, refer to [Enable FIPS mode](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening) in Red Hat's documentation. To verify that FIPS mode is enabled, refer to [Verify that the kernel enforces FIPS mode](#verify-that-the-kernel-enforces-fips-mode).
+
+### Install on Red Hat Enterprise Linux
+
+When you install Red Hat Enterprise Linux (RHEL) on a host, or after installation has completed, you can [enable FIPS mode](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening). Red Hat recommends installing RHEL with FIPS mode enabled, as opposed to enabling FIPS mode later to help ensure that the system generates all keys with FIPS-approved algorithms and continuous monitoring tests in place.
+
+After FIPS mode is enabled, when you install OpenSSL, FIPS-validated OpenSSL libraries are installed. To install OpenSSL on RHEL:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+microdnf -y install openssl crypto-policies-scripts
+fips-mode-setup --enable
+~~~
+
+Next, [verify that the kernel enforces FIPS mode](#verify-that-the-kernel-enforces-fips-mode).
+
+### Verify that the kernel enforces FIPS mode
+
+If FIPS mode is not enabled in the Linux kernel, or if OpenSSL is not configured correctly, it does not enforce FIPS 140-2 by default even if the FIPS module is present. Take these steps to verify that the Linux kernel is enforcing FIPS 140-2:
+
+1. You can list the FIPS-validated ciphers using the `openssl ciphers` command. If FIPS mode is disabled, a `no cipher match` error occurs.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    openssl ciphers FIPS -v
+    ~~~
+
+1. When FIPS mode is enabled, the `sysctl` variable `fips_enabled` is set to `1`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    sysctl crypto.fips_enabled
+    ~~~
+
+    ~~~ shell
+    1
+    ~~~
+
+    {{site.data.alerts.callout_info}}
+    This command does not work from within a Docker container. A Docker container runs with FIPS mode enabled if the Docker host has FIPS mode enabled. Run this command from the Docker host rather than the Docker container.
```

Comment:
```suggestion
    The preceding command does not work from within a Docker container. A Docker container runs with FIPS mode enabled if the Docker host has FIPS mode enabled. Run this command from the Docker host rather than from the Docker container.
```